packages/cli/src/controllers/folder.controller.ts

```diff
@@ -67,4 +67,23 @@ export class ProjectController {
 			throw new InternalServerError();
 		}
 	}
+
+	@Delete('/:folderId')
+	@ProjectScope('folder:delete')
+	async deleteFolder(
+		req: AuthenticatedRequest<{ projectId: string; folderId: string }>,
+		_res: Response,
+		@Body payload: DeleteFolderDto,
+	) {
+		const { projectId, folderId } = req.params;
+
+		try {
+			await this.folderService.deleteFolder(folderId, projectId, payload);
+		} catch (e) {
+			if (e instanceof FolderNotFoundError) {
```

Comment:
`FolderNotFoundError` could extend `NotFoundError` so this whole try catch would be unnecessary

Reply:
I did not want to attach the exception to an HTTP error as `deleteFolder` can be used in a controller or outside of it (e.g the CLI). Using this outside of a controller and getting an HTTP error would be weird. Thoughts?

Reply:
Yeah it's valid question. I have personally started to ignore this leakage of concerns between layers when it comes to errors between the http and inner layers. In practice it doesn't usually matter if the error has some extra fields related to HTTP. But this is not something I feel strongly about, so I'm perfectly happy also to go with this option 👍